binaries/workerserver/main.go

```diff
@@ -8,7 +8,8 @@ import (
 
 	"github.com/apache/thrift/lib/go/thrift"
 	"github.com/scootdev/scoot/common/endpoints"
-	"github.com/scootdev/scoot/runner/execer/fake"
+	"github.com/scootdev/scoot/runner/execer/execers"
+	osexec "github.com/scootdev/scoot/runner/execer/os"
```

Comment:
I lean towards folding /os into the /execers since it's not all that
complicated. For more complicated impls, I imagine they'll live in an
entirely different directory (i.e. not in runner/execer), particularly due
to the twitter/oss divide. Ultimately it's preference so I don't have a
convincing argument for either way.

On Tue, Sep 13, 2016 at 4:15 PM, Dan Bentley <notifications@github.com>
wrote:

> In binaries/workerserver/main.go
> <https://github.com/scootdev/scoot/pull/99#discussion_r78662144>:
>
> > @@ -8,7 +8,8 @@ import (
> >
> >  	"github.com/apache/thrift/lib/go/thrift"
> >  	"github.com/scootdev/scoot/common/endpoints"
> > -	"github.com/scootdev/scoot/runner/execer/fake"
> > +	"github.com/scootdev/scoot/runner/execer/execers"
> > +	osexec "github.com/scootdev/scoot/runner/execer/os"
>
> In general, we *can't* have every implementation in any one package. And
> that's part of the point of golang: implementations don't have to be
> related to the interface.
>
> But I think Caitie made the perfectly cromulent point that for many
> interfaces, there are simple impls that are not fake but want to live
> together. I think for these, it's good to lump them together.
>
> I think execer/os is right on the line. But, e.g., I think a git-based
> implementation of snapshot is too complex, and should be in its own package.
>
> WDYT?
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/scootdev/scoot/pull/99/files/b19782b475528cb9abf63eb3e608c638a413b651#r78662144>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABc5CYB6_3fIWPhOyGNStRqBYG78rsSzks5qpy6egaJpZM4J3orw>
> .
>